src/crawler.js

```diff
@@ -61,7 +65,7 @@ function normalizeIpp(ip, port) {
 
     out_ip = splitIp
     out_port = port || splitPort || DEFAULT_PORT
-    ipp = out_ip + ':' + out_port
+    var ipp = out_ip + ':' + out_port
   }
 
   return ipp
```

Comment:
I think it'd be better to raise an exception than return undefined.

Raising an exception is a very disruptive thing that forces the API consumer to sit in the corner and think about what they've done since the whole program will explode.

Returning undefined means that the API user needs to remember to check for a valid return value from a function. In this particular instance, a function called ``normalizeIpp`` implies that it'll return a normalized IP and port in the form of a string. Returning undefined breaks that assumption.

tl;dr Exceptions are explicit error reporting, returning undefined values is implicit error reporting, and programmers really hate having to remember things.